lua/nvim-lsp-installer/ui/display.lua

```diff
@@ -200,30 +240,34 @@ function M.new_view_only_win(name)
                 draw(renderer(new_state))
             end)
 
+            -- we don't need to subscribe to state changes until the window is actually opened
+            unsubscribe(true)
+
             return mutate_state, get_state
         end,
         open = vim.schedule_wrap(function(opts)
             -- log.debug { "opening window" }
             assert(has_initiated, "Display has not been initiated, cannot open.")
-            local win = vim.fn.win_findbuf(buf)[1]
-            if win and vim.api.nvim_win_is_valid(win) then
+            if win_id and vim.api.nvim_win_is_valid(win_id) then
+                -- window is already open
                 return
             end
             unsubscribe(false)
-            local opened_win = open(opts)
+            local opened_win_id = open(opts)
             draw(renderer(get_state()))
-            redraw_by_winnr[opened_win] = function()
+            redraw_by_win_id[opened_win_id] = function()
                 draw(renderer(get_state()))
+                vim.api.nvim_win_set_config(opened_win_id, create_popup_window_opts())
```

Comment:
Here's the redraw logic. We should try to see if we can simplify it a bit as to not have a chance of an invalid `win_id`?

cc @Tastyep

Reply:
Maybe checking `nvim_win_is_valid()`?

Reply:
@williamboman, we should probably introduce a lot of debug prints in this file. It would help with this and in the future.